src/eureka/S3_data_reduction/s3_reduce.py

```diff
@@ -696,7 +696,11 @@ def reduce(eventlabel, ecf_path=None, s2_meta=None, input_meta=None):
             else:
                 normspec = util.normalize_spectrum(meta, data.aplev.values)
                 meta.mad_s3 = util.get_mad_1d(normspec)
-            log.writelog(f"Stage 3 MAD = {int(np.round(meta.mad_s3))} ppm")
+            try:
+                log.writelog(f"Stage 3 MAD = {int(np.round(meta.mad_s3))} ppm")
+            except:
+                log.writelog("Could not compute Stage 3 MAD")
```

Comment:
Might be worth adding WARNING at the start of this line since a NaN MAD value isn't a good sign